MapboxNavigation/NavigationViewController.swift

```diff
@@ -319,9 +319,10 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
 
         mapViewController?.notifyDidChange(routeProgress: routeProgress, location: location, secondsRemaining: secondsRemaining)
         tableViewController?.notifyDidChange(routeProgress: routeProgress)
-        
-        if routeProgress.currentLegProgress.alertUserLevel == .arrive {
-            navigationDelegate?.navigationViewController?(self, didArriveAt: destination)
+
+        if routeProgress.currentLegProgress.alertUserLevel == .arrive,
+            routeProgress.remainingWaypoints.count == 0 {
```

Comment:
Should the developer be notified when the user reaches an intermediate waypoint? I could see use cases for only responding to the final arrival and for responding to intermediate arrivals as well. Perhaps the delegate method should have an additional argument, `numberOfWaypointsRemaining`, to allow the developer to distinguish between the two cases.